go/arrow/datatype_fixedwidth.go

```diff
@@ -325,6 +325,22 @@ func (u TimeUnit) Multiplier() time.Duration {
 
 func (u TimeUnit) String() string { return [...]string{"s", "ms", "us", "ns"}[uint(u)&3] }
 
+var _strToTimeUnit = map[string]TimeUnit{
+	"s":  Second,
+	"ms": Millisecond,
+	"us": Microsecond,
+	"ns": Nanosecond,
+}
+
+// TimeUnitFromString is the reverse of TimeUnit.String
+func TimeUnitFromString(s string) (TimeUnit, error) {
+	if v, ok := _strToTimeUnit[s]; ok {
+		return v, nil
+	}
+
+	return 0, fmt.Errorf("unsupported time unit: %q", s)
+}
```

Comment:
is this in use anymore? Is it worthwhile to keep?

Reply:
Not used anymore.
I'll remove this in a minute

Reply:
Removed in 2bf65c58573f6ddbdc2e0f6fed13dd79516e6bac